Dockerfile

```diff
@@ -1,30 +1,30 @@
+#Base image from CI
 FROM registry.ci.openshift.org/ocp/4.11:base
-
 ARG KERNEL_VERSION=''
 ARG RT_KERNEL_VERSION=''
 ARG RHEL_VERSION=''
-
+ARG RT_ENABLED
 RUN echo ${RHEL_VERSION} > /etc/yum/vars/releasever \
     && yum config-manager --best --setopt=install_weak_deps=False --save
 
-# kernel packages needed to build drivers / kmods 
-RUN yum -y install \
-    kernel-core${KERNEL_VERSION:+-}${KERNEL_VERSION} \
-    kernel-devel${KERNEL_VERSION:+-}${KERNEL_VERSION} \
-    kernel-headers${KERNEL_VERSION:+-}${KERNEL_VERSION} \
-    kernel-modules${KERNEL_VERSION:+-}${KERNEL_VERSION} \
-    kernel-modules-extra${KERNEL_VERSION:+-}${KERNEL_VERSION} \
-    && yum clean all
-
-# real-time kernel packages
-RUN if [ $(arch) = x86_64 ]; then \
+# For real-time kernel packages use --build-arg RT_ENABLED=1 
+RUN if [ $(arch) = x86_64 ] && [[ -n "$RT_ENABLED" ]]; then \
```

Comment:
Currently ART is building 2 images per release (AFAIK):
* x86_64 image
* ARM image

This change means that the ART team has to now build 3 images for each release:
* x86_64 image with standard kernel
* x86_64 image with rt kernel
* ARM image

If we decide to go with this, we need to make sure to be synced with the ART team.

@dagrayvid WDYT?

Reply:
@enriquebelarte, as this is currently written, the standard kernel will never be installed unless `[ $(arch) = x86_64 ]`. Unless I'm missing something, this will completely break DTK on non-x86 architectures. This doesn't seem to be the intended change. 

If you want to make changes to how different architectures are built, or how realtime kernels are included, I suggest that you split these into different PRs. Removing KVC is a big enough change for one PR.

Currently DTK is built for ARM, s390x, pp64le, and x86_64. For instance see the latest build in brew which includes all 4 architectures: https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=1983634

Has this idea been talked about with ART? I'm not sure if it's possible for them to ship a completely separate image for RT enabled. Why do we need/want this to be a separate image?

Reply:
The idea with including the RT kernel in DTK was that all of the needed dependencies for kmods would be available in the one base image. DTK is currently shipped in the payload, and adding another version of DTK to the payload would probably be rejected by the OCP architects. Is the idea that this Dockerfile would be used in buildconfigs only, for on-cluster builds of a custom RT DTK?

Reply:
> If you want to make changes to how different architectures are built, or how realtime kernels are included, I suggest that you split these into different PRs. Removing KVC is a big enough change for one PR.

I absolutely agree with that, let's just handle the KVC detaching in this PR.

> Has this idea been talked about with ART?

Not yet, I don't think we will go that direction. @enriquebelarte let's keep it the way it is today (in x86 we install the "normal" package and also the RT kernel packages (although we will have redundant packages).

> The idea with including the RT kernel in DTK was that all of the needed dependencies for kmods would be available in the one base image.

Yea, I think we are going to keep that behaviour for now.

> Why do we need/want this to be a separate image?

We don't, I just commented on this PR that this is breaking the current behaviour so we should stick with the original design for now.